### ISSUE TITLE
submission to CRAN or bioconductor?

### ISSUE THREAD

**Issue by nick-youngblut** [jbisanz/qiime2R#3]
Thanks for making this package! It beats running a long set of command line commands to do the conversion. 

As far as I can tell, `qiime2R` isn't available on CRAN or bioconductor as of yet, although your recent git commit message "v0.99 preparing for bioconductor" suggests that `qiime2R` will soon be on bioconductor. When do you think that your package will be available on bioconductor?

I'm asking because I'd like to create a bioconda recipe for `qiime2R`, but it is *much* preferred to have a CRAN or bioconductor url instead of a GitHub release url.

**Comment by jbisanz** [comment]
Hi Nick,

This is certainly in my to do list but I have been occupied with those pesky wet lab experiments. I will reply again when it has been done. 

Jordan

**Comment by nick-youngblut** [comment]
Thanks for letting me know. Good luck with the benchwork!

**Comment by colinbrislawn** [comment]
Good evening,

I would like to see this on bioconda as well. Any plans for CRAN or bioconductor in 2023?

**Comment by antagomir** [comment]
Thanks @colinbrislawn - we are thinking the same. Availability in CRAN or Bioconductor would be a must for linking this to the vast number of quality-controlled packages in these ecosystems. The qiime2R package contains useful functionality but we cannot take advantage of this as a dependency in Bioconductor packages because Bioc guidelines require that the package is available via CRAN or Bioc. 

Any updates on the plans, or possibilities to help on this?

**Comment by jbisanz** [comment]
I started the process and then it fell off my radar. I will have some spare time next week to hopefully get this submitted. @antagomir @colinbrislawn any preference for cran/bioconductor?

**Comment by colinbrislawn** [comment]
CRAN I guess? No strong preference from me. Either way we wrap it in a conda package 📦  and put submit it to bioconda
https://github.com/bioconda/bioconda-recipes 

I suppose the other option is to skip the standard repos and go directly to bioconda. This seems bad, though.

**Comment by antagomir** [comment]
Ok, some remarks:

- CRAN is a bit more light-weight in terms of formal requirements and automated checks, and packages get accepted faster when they pass the automated tests
- Bioconductor has more stringent requirements but it also comes with remarkable visibility + explicit focus on life science applications. In principle, this would be a better choice overall and it seems that the package was already somewhat prepared for Bioc submission previously. 

CRAN or Bioc both guarantee certain minimum standards for software quality, documentation and cross-platform compatibility. Imo it is worth the effort. 

Additional impact will come from the fact that after this other CRAN/Bioc packages (like ours) can use this qiime2R as a dependency.  Otherwise, we will be forced to create parallel implementations of these very useful tools (and license compatibility issues won't allow borrowing the code as we are all using different open licenses..).

**We can provide some support** if you are willing to pull it through to either of these repositories. The automated checks will indicate what needs to be done, and in most cases it will take just some routine coding to get it through. After this, minimum maintenance is usually needed for the years to come.

**Comment by antagomir** [comment]
Hi! Any updates on this? 

It would be really useful to have this in CRAN or Bioconductor in order to utilize the tools in other CRAN/Bioc pkgs.

**Comment by antagomir** [comment]
Using this as dependency in CRAN or Bioconductor pkgs depends on availability via CRAN; Conda version does not solve that issue.

**Comment by TuomasBorman** [comment]
I agree that the package should be submitted to CRAN. It is rather simple and would benefit also large Bioconductor community.

**Comment by antagomir** [comment]
Would be informative to hear if there are plans towards this? Solving the checks and submitting to CRAN is relatively straightfwd process and it will improve broad accessibility.
